source/DefaultDocumentation.Test/Markdown/FileNameFactories/Md5FactoryTest.cs

```diff
@@ -10,6 +10,6 @@ public sealed class Md5FactoryTest : AFileNameFactoryTest<Md5Factory>
         public void Name_Should_return_FullName() => Check.That(Name).IsEqualTo("Md5");
 
         [Fact]
-        public void GetFileName_Should_return_Md5() => Test(AssemblyInfo.ClassDocItem, "07ucmh7NagP7qlcO1w9snQ.md");
+        public void GetFileName_Should_return_Md5() => Test(AssemblyInfo.ClassDocItem, "NFMDH6WUUDQ1ZFSD9PKB7FIB9.md");
```

Comment:
I think it would be nice to add a couple test cases here for the base36 encoding routine, especially around the logic to ensure non-negative and the logic to ensure we don't return an empty string.

However, this would require adding `[assembly: InternalsVisibleTo("DefaultDocumentation.Tests")]` to the DefaultDocumentation.Markdown project. Would that be OK?

Reply:
Sorry for the late reply! I am not a fan of `InternalsVisibleTo` but I get that you would want to secure the implementation with some tests. In essence the interface of IFileNameFactory should act almost as a passthrough anyway (the context is not really used and it will probably only use the `DocItem.FullName`), the `NamespaceDocItem` is probably the easiest implementation of `DocItem` to use to pass whatever you want. If it's really too cumbersome just use `InternalsVisibleTo` :).

Also sorry it seems the latest visual studio update added some analyzers that make the ci fails, I'll check to update the rules I want to keep!

Reply:
@Doraku ok I've added a couple new tests using the mechanism you mentioned.

Reply:
I fixed the warning as error, if you rebase on master the ci should run the tests correctly now.

Reply:
Thanks! Checks are passing now.